src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,134 @@
+package frc.trigon.robot.subsystems.arm;
+
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Arm extends SubsystemBase {
+
+    private final static Arm INSTANCE = new Arm();
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    private final CANSparkMax
+        masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+        followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+        masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+        followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public Command getSetTargetAngleCommand(Rotation2d angle){
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(angle),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {},
+                () -> false,
+                this
+        );
+    }
+
+    public Command getSetTargetElevatorCommand(Rotation2d angle){
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(angle),
+                this::setTargetElevatorFromProfile,
+                (interrupted) -> {},
+                () -> false,
+                this
+        );
+    }
+
+    private void setTargetAngleFromProfile(){
+        if (angleMotorProfile == null){
+            stopAngleMotors();
+            return;
+        }
+
+        final TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        masterAngleMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
+    }
+
+    private void setTargetElevatorFromProfile(){
+        if (elevatorMotorProfile == null){
+            stopElevatorMotors();
+            return;
+        }
+
+        final TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        masterElevatorMotor.getPIDController().setReference(targetState.position, CANSparkMax.ControlType.kPosition);
```

Comment:
again, you can't use `getPIDController().setReference` since that uses the onboard pid of the sparkmax. You can't use the onboard pid since the encoder isn't from the same company as the motor. So you'll have to use the pid controllers you created earlier to calculate the power you need to input to the motor, and give that as the power to the motor.

Reply:
what pid controllers I made earlier?

Reply:
In the constants